### PR TITLE
op-chain-ops: Improve OVM_ETH migration, remove invalid check slots

### PR DESCRIPTION
**Pull request by mslipper**
When we ran the migration rehearsal, the OVM_ETH migration took approximately an hour and a half. This was because the storage slots in the OVM_ETH contract were iterated twice: once to check them, and another time to actually migrate them. To reduce the amount of time taken, this PR updates the OVM_ETH migration to check and migrate in one step. This does not cause database corruption on failure, since the trie changes are buffered in memory and aren't committed until all the checks pass. Total migration time should be reduced by approximately 40 minutes.

Separately, I removed the invalid check slots referring to the EIP-1967 slots on the L2XDM. These slots no longer exist, and were causing the post-check steps to fail.

Closes CLI-3590
Closes CLI-3576
Closes https://github.com/ethereum-optimism/optimism/issues/5048

### CODE REVIEW

**Comment by mslipper** [comment]
Marking as draft while I benchmark.

File: op-chain-ops/genesis/check.go [hunk 1]
```diff
@@ -59,9 +59,6 @@ var (
 			common.Hash{31: 0x97}: common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000001"),
 			// Slot 0xcc (204) is xDomainMsgSender
 			common.Hash{31: 0xcc}: common.HexToHash("0x000000000000000000000000000000000000000000000000000000000000dead"),
-			// EIP-1967 storage slots
-			AdminSlot:          common.HexToHash("0x0000000000000000000000004200000000000000000000000000000000000018"),
```
Comment (tynes) [inline op-chain-ops/genesis/check.go:63]: The messenger is still behind a proxy - its 51 and 151 that no longer exist

See:
- https://github.com/ethereum-optimism/optimism/blob/4071f64e5dd582beaa28c393b20757193361a4ca/packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol#L77
- https://github.com/ethereum-optimism/optimism/blob/4071f64e5dd582beaa28c393b20757193361a4ca/packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol#L45

**Review by Balochistan-has — COMMENTED** [review]
https://github.com/ethereum-optimism/optimism/pull/5106/files#diff-c08feda74e0f1bce60c0ed2b207a0d29895c7a326d6172b2c8a7a26f5cc30114

File: op-chain-ops/genesis/db_migration.go [hunk 2]
```diff
@@ -192,10 +182,15 @@ func MigrateDB(ldb ethdb.Database, config *DeployConfig, l1Block *types.Block, m
...
+	// We also need to verify that we have all of the storage slots for the LegacyERC20ETH contract
+	// that we expect to have. An error will be thrown if there are any missing storage slots.
+	// Unlike with withdrawals, we do not need to filter out extra addresses because their balances
+	// would necessarily be zero and therefore not affect the migration.
+	//
+	// Once verified, we migrate the balances held inside the LegacyERC20ETH contract into the state trie.
 	// We also delete the balances from the LegacyERC20ETH contract.
 	log.Info("Starting to migrate ERC20 ETH")
-	err = ether.MigrateLegacyETH(db, addrs, int(config.L1ChainID), noCheck)
+	err = ether.MigrateLegacyETH(db, migrationData.Addresses(), migrationData.OvmAllowances, int(config.L1ChainID), noCheck, commit)
```
Comment (tynes) [inline op-chain-ops/genesis/db_migration.go:193]: I want to make sure there is no diff passing in `migrationData.Addresses()` compared to `addrs` which is returned from `PreCheckBalances`

https://github.com/ethereum-optimism/optimism/blob/35d4a37c2b9a59e9bf16fa4c5e22283f2348c3d2/op-chain-ops/ether/precheck.go#L18

Comment (tynes) [reply op-chain-ops/genesis/db_migration.go:193]: It looks good based on the new logic

Comment (mslipper) [reply op-chain-ops/genesis/db_migration.go:193]: The OVM ETH migration only migrates what's actually in state. It doesn't create any new state like the withdrawals migration does, so we don't need to filter the input like we do with withdrawals.

**Review by tynes — APPROVED** [review]

**Pull request merged**